### ISSUE TITLE
Add a way to call the default tool behaviour when on_use is defined.

### ISSUE THREAD

**Issue by kromka-chleba** [minetest/minetest#12917]
## Problem
TL;DR on_use callback overwrites the default behaviour of a tool (digging a node) and it can't be accessed
I'm trying to add a multipurpose hoe to a game that tills soil and creates a particle spawner if a node is punched 3 times but digs the node normally if LMB is clicked and held (the default behaviour).
To solve this problem I added on_use callback to the tool and found out that it overwrites the default behaviour - the hoe tills soil as I want, but tool capabilities get ignored (punch interval is ignored, the tool can't give damage or dig nodes), which is undesirable.
According to the API documentation the default on_use is nil.
There is an unresolved issue on the forums about this, but I can't find it again because searching for "on_use" complains that "on" and "use" are too common.

## Solution
Please make it possible to access the default tool on_use behaviour somehow.

## Alternatives
I considered setting on_punch for all soils in the game, but this isn't scalable.

**Comment by fluxionary** [comment]
> I considered setting on_punch for all soils in the game, but this isn't scalable.

a scalable solution could be created via `register_on_punchnode`:
https://github.com/minetest/minetest/blob/8f1593e4e8bef498b7021e330e3558545c9543ee/doc/lua_api.txt#L5160-L5161

for your specific use-case, you could define an `on_place` callback for the tilling behavior (assuming you don't already have one):
https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L8089-L8095

**Comment by sfan5** [comment]
The reason you can't just "call" the default behaviour is that it's handled on the client while mods execute on the server. There's also the latency, it may very well be too late to seamlessly invoke the default behaviour by that time.

As mentioned you might be able to use `register_on_punchnode` (which is _not_ the same as `on_punch`) create a solution that works for all nodes.

**Comment by kromka-chleba** [comment]
> The reason you can't just "call" the default behaviour is that it's handled on the client but mods execute on the server. There's also the latency, it may very well be too late to seamlessly invoke the default behaviour by that time.
> 
> As mentioned you might be able to use `register_on_punchnode` (which is _not_ the same as `on_punch`) create a solution that works for all nodes.
Okay, so `register_on_punchnode` registers a function that's called for every punched node in the game, right?
If so then this mostly solves the issue, though my initial solution had one advantage - the hoe created a timer and a particle spawner only for the node that was punched, so other nodes didn't have any on_punch registered.
As the other person suggested it would be nice to have something like "on_tap", because "on_use" kills tool capabilities entirely. I really could rewrite the default behaviour in Lua, but that's not the most interesting thing to do.

Anyway, I think you can close the issue, yet would be nice if there was a callback for that.
